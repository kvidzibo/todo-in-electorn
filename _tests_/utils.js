import { shallow, configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

beforeAll(() => {
  configure({ adapter: new Adapter() })
})

export const render = (comp, mode = 3) => {
  switch (mode) {
    case 1:
      return shallow(comp)
    case 2:
      return renderer.create(comp).toJSON()
    case 3:
      return {
        component: shallow(comp),
        tree: renderer.create(comp).toJSON()
      }
    case 4:
      return mount(comp)
    case 5:
      return {
        component: mount(comp),
        tree: renderer.create(comp).toJSON()
      }
  }
}
