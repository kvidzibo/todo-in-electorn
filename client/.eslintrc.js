module.exports = {
    "extends": "standard",
    "globals": {$: true},
    "plugins": [
      "react"
    ],
    "parser": "babel-eslint",
    "rules": {
      "react/jsx-uses-vars": 1,
      "react/jsx-uses-react": 1
    }
};
