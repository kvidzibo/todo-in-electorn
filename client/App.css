.ml5 {
    margin-left: 0.5rem;
}
