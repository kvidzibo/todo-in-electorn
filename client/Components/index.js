export {default as Hero} from './Hero'
export {default as Section} from './Section'
export {default as Field} from './Field'
export {default as Button} from './Button'
