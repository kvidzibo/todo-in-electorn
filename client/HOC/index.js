export {default as withRefetch} from './withRefetch'
