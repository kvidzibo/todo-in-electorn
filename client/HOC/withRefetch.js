import React, {PureComponent} from 'react'

export default function withRefetch (WrappedComponent, Module, fetch) {
  return class extends PureComponent {
    state = {
      data: ''
    }
    componentDidMount () {
      Module.on('updated', this.fetch)
      Module.on('inserted', this.fetch)
      Module.on('removed', this.fetch)
      this.fetch()
    }

    componentWillUnmout () {
      console.log('lol')
      Module.removeEventListener('updated', this.fetch)
      Module.removeEventListener('inserted', this.fetch)
      Module.removeEventListener('removed', this.fetch)
    }

    fetch = () => {
      fetch().then(data => {
        this.setState({data})
      })
    }

    render () {
      return <WrappedComponent data={this.state.data} {...this.props} />
    }
  }
}
