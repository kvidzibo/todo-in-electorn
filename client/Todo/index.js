export {default as List} from './List'
export {default as Module} from './Module'
