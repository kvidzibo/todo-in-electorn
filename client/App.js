import React, { PureComponent } from 'react'

import 'bulma/css/bulma.css'
import 'font-awesome/css/font-awesome.min.css'
import './App.css'

import { Hero, Section } from './Components'
import { withRefetch } from './HOC'
import { List as TodoList, Module as TodoM } from './Todo'

class App extends PureComponent {
  render () {
    const { data } = this.props

    return (
      <div className="app">
        <Hero
          title="Todo in electron"
          subtitle="The best todo app ever!"
          primary
        />
        <Section title="List" subtitle={`todo: ${data[1]} done: ${data[0]}`}>
          <TodoList />
        </Section>
      </div>
    )
  }
}

export default withRefetch(App, TodoM, () => Promise.all([
  TodoM.count({status: 'done'}),
  TodoM.count({status: 'todo'})
]))
