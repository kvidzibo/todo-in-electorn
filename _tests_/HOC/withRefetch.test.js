import React from 'react'
import { render } from '../utils'
import { withRefetch } from '../../client/HOC'

let ModuleMock
let fetch
let WithRefetch

beforeEach(() => {
  ModuleMock = {on: jest.fn(), removeEventListener: jest.fn()}
  fetch = jest.fn().mockResolvedValue('')
  WithRefetch = withRefetch(
    'h1',
    ModuleMock,
    fetch
  )
})

test('adds listeners and calls fetch', () => {
  render(<WithRefetch />, 1)
  expect(ModuleMock.on.mock.calls.length).toBe(3)
  expect(fetch.mock.calls.length).toBe(1)
  const tree = render(<WithRefetch />, 2)
  expect(tree).toMatchSnapshot()
})

test('passes data', () => {
  const component = render(<WithRefetch />, 1)
  expect(component.find('h1').props().data).toBe('')
})
