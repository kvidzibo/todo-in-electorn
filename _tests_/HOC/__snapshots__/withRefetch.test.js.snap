// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds listeners and calls fetch 1`] = `
<h1
  data=""
/>
`;
